perfect_division.py

```diff
@@ -0,0 +1,27 @@
+from math import sqrt
+
+
+def div(number_to_check):
+    for i in range(1, int((number_to_check / 2)) + 1):
+        if number_to_check % i == 0:
+            yield i
+
+
+def check_for_perfection(number_to_check):
+    Sum = 0
+    generator = div(number_to_check)
+    for number in generator:
+        Sum += number
+    if Sum > number_to_check:
+        return False
+    elif Sum == number_to_check:
+        return number_to_check
+
```

Comment:
Are you sure this is the best implementation for the code? (is there a way to shorten and to simplify it?)